### DISCUSSION TITLE
When a provider [autoDispose] is destroy?

### DISCUSSION THREAD

**Discussion by quangson91** [rrousselGit/riverpod#3099] [Q&A]
I already read the docs: https://riverpod.dev/docs/concepts/provider_lifecycles

which says that when provider is no longer used -> the autodispose provider will be destroy.

But I am still confuse.

Here is example:

Given: `myProvider` which is an autodispose provider.

Now I have two screen A, B;

When user click an button from screen A -> I want to init `ref.read(myProvider).init()` then open screen B.

My expectation is `myProvider` will live along with screen `B`. 

The problem need to be clear:

1. If in screen `B` I do not call/use `myProvider` (eg: ref.listen, ref.watch myProvider) => Is that `myProvider` still alive ?

2. if in the `build()` of screen `B` I call `ref.listen(myProvider)` => It seems `myProvider` will be alive along with screen `B`, right ?
But, I am not sure the instance of `myProvider` when we call `ref.read(myProvider).init()` & the provider here are the same ?
I mean, is there anycase that the lifecycle of `myProvider` looks like:

```
-> ref.read(myProvider).init()
-> myProvider create & alive
-> ... some how it will be destroy before we listen in screen B.
-> ref.listen(myProvider) in screen B -> new instance will be create & alive.
```

If there anything unclear, please tell me.

Thanks!

**Comment by AhmedLSayed9** [comment]
That's expected.

Navigating takes time and during that time your provider is disposed.

Instead of calling `.init()`, your provider should have the initialization in its build method.

**Comment by meroo36** [comment]
Please check [this.](https://riverpod.dev/docs/essentials/do_dont#avoid-initializing-providers-in-a-widget). You shouldn't call init(). the provider needs to initialize itself as the documentation recommends.

Your case makes sense to me since you are not watching/listening to the provider on screen A. So it would make sense if it is being disposed right after you call the init() since it's not being used.
